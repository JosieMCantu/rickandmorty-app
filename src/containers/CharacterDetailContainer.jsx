import React, { Component } from 'react';
import { characterById } from '../services/rickAndMortyApi';
import CharacterDetail from '../components/characters/CharacterDetail';


export default class CharacterDetailContainer extends Component {
    state = {
        character: {},
    }

    componentDidMount = async () => {

        const { id } = this.props.match.params

        const character = await characterById(id);

        if (character.id === id) {
            return character;
        }

        this.setState({character});
    }


    render() {

        const { character } = this.state;
        
        return (
            <div>
                <CharacterDetail character={character} />
            </div>
        )
    }
}

