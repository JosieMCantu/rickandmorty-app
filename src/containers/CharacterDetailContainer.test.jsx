import React from 'react';
import { render, screen } from '@testing-library/react';
import CharacterDetailContainer from './CharacterDetailContainer';

describe('Character Detail Container', () => {
    it('renders one character', async () => {

        render(<CharacterDetailContainer match={{params:
        {character: ''}}} />);

        screen.getByText('Detail Page');
        
        const ul = await screen.findByRole('list', { name: 'character' });
        expect(ul).toMatchSnapshot();
    });
});

