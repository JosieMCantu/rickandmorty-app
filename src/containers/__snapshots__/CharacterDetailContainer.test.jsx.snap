// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character Detail Container renders one character 1`] = `
<ul
  aria-label="character"
>
  <li>
    <h1>
      Detail Page
    </h1>
    <p />
    <p />
    <p />
    <p />
    <img />
  </li>
</ul>
`;
